trin-beacon/src/network.rs

```diff
@@ -56,14 +57,16 @@ impl BeaconNetwork {
         if portal_config.trusted_block_root.is_some() {
             tokio::spawn(async move {
                 let beacon_sync = BeaconSync::new(overlay_tx);
-                beacon_sync
+                let beacon_sync = beacon_sync
                     .start(
                         portal_config
                             .trusted_block_root
                             .expect("Trusted block root should be available"),
                     )
-                    .await
-                    .expect("Beacon sync failed to start");
+                    .await;
+                if let Err(e) = beacon_sync {
+                    error!("Failed to start beacon sync: {:?}", e);
```

Comment:
*nitpick* interpolate `e` into the string... though clippy lint should be picking up on this?